### DISCUSSION TITLE
More articles?

### DISCUSSION THREAD

**Discussion by spelled-ayayron** [gilzoide/godot-lua-pluginscript#37] [General]
Hi, I've been following along with the articles in this repo about implementing Lua since I'm trying to implement a [Luau](https://github.com/Roblox/luau/) GDNative + PluginScript library for my own experimentation/use case. The series seems to end and hasn't been updated in a while and I was wondering if they were going to get continued or if I should continue my hunt for information elsewhere. It's my first time really working in C and Godot and I haven't really found anywhere else that offers as much of an in-depth tutorial on implementing a new language like here.

**Comment by gilzoide** [comment]
Hey @seyaidev!
Well, yeah, I've left the making off articles frozen for a long time now and finishing those are not in my priorities, sorry =/
Even when I was writing them, the actual repo code was already so much more advanced, that I had to create the `blog-code` branch to keep the code written in the articles. This is not easily maintainable and there was just so much to fix in the actual code and stuff, that I kind of abandoned the devlog thing.

When I was making the Lua PluginScript I didn't really find much information either, there were some few things aroung GDNative per se and that was it, so I know how you feel. Reading Godot code, godot-cpp and godot-python helped a lot.

But hey, either way, I'll be more than happy to help you with information and ideas around Godot + GDNative + PluginScript, just keep posting here, in a new discussion or hit me up via email or any other channel of preference, just let me know =]

Say, do you want to support Godot 3.X or 4.0? The native extensions scheme changed a lot between them.
I haven't used GDExtensions yet, but am starting to look into it to port Lua PluginScript to Godot 4.0. A lot has changed, but the C stuff is simpler and seems stable enough, and the C++ repo is up to date.

**Reply by spelled-ayayron** [reply]
Thanks for the reply. Totally understand why it ended up on the backburner. I'll take a look at reading godot-cpp and godot-python soon as well. I did see that there is the new GDExtension being introduced with Godot 4.0 so I'll be walking into uncharted territory it seems with implementing Luau. I'll be back again when I run into a roadblock! 😅

**Reply by gilzoide** [reply]
Hi again @seyaidev!
As I just posted in https://github.com/gilzoide/godot-lua-pluginscript/issues/19#issuecomment-1374589350, I was looking into how porting Lua PluginScript to a GDExtension for Godot 4.0 support would go.
If Luau has a C API compatible with Lua's, my idea of generating Lua/C bindings might help you out, at least with the native bindings. Or maybe the whole thing will be compatible, and changing Lua/LuaJIT for Luau will do the trick 🤔
It's quite a rewrite, but now it seems like a better approach than the one I had for GDNative.

If I start reimplementing Lua support with GDExtensions, I'll try to write up-to-date articles for every step. New year, new resolutions to possibly fail, but hopefully not 👀

> I'll take a look at reading godot-cpp and godot-python soon as well

Feel free to also look into this repo's code, if it helps anything... The code order and build process are not that well documented, but if you know Lua and maybe LuaJIT's FFI, it should be easy enough looking into the Lua source, which is responsible for almost everything in the implementation.
The order of the scripts is in the [Makefile](https://github.com/gilzoide/godot-lua-pluginscript/blob/main/Makefile#L77-L120). They're all glued together by the building system, that's why local variables are accessed in other files. LSPs go like "¿¿¿what the???".

> I'll be back again when I run into a roadblock!

You don't need to go that far for calling out, don't worry. Do as you feel is best.
I don't promise to answer quickly, but I should answer everything eventually.
